src/tools/tidy/src/deps.rs

```diff
@@ -48,39 +48,74 @@ const LICENSES: &[&str] = &[
 
 type ExceptionList = &'static [(&'static str, &'static str)];
 
+pub(crate) struct WorkspaceInfo<'a> {
+    /// Path to the directory containing the workspace root Cargo.toml file.
+    pub(crate) path: &'a str,
+    /// The list of license exceptions.
+    pub(crate) exceptions: ExceptionList,
+    /// Optionally:
+    /// * A list of crates for which dependencies need to be explicitly allowed.
+    /// * The list of allowed dependencies.
+    pub(crate) crates_and_deps: Option<(&'a [&'a str], &'a [&'a str])>,
+    /// Submodules required for the workspace
+    pub(crate) submodules: &'a [&'a str],
+}
+
+impl<'a> WorkspaceInfo<'a> {
+    const fn new(
+        path: &'a str,
+        exceptions: ExceptionList,
+        crates_and_deps: Option<(&'a [&str], &'a [&str])>,
+        submodules: &'a [&str],
+    ) -> Self {
+        Self { path, exceptions, crates_and_deps, submodules }
+    }
+}
```

Comment:
Suggestion: maybe inline so the construction-site is more like

```rs
WorkspaceInfo {
    path: ".",
    exceptions: EXCEPTIONS,
    crates_and_deps: Some((&["rustc-main"], PERMITTED_RUSTC_DEPENDENCIES)),
    submodules: &[],
},
```

and less like

```rs
WorkspaceInfo::new(".", EXCEPTIONS, Some((&["rustc-main"], PERMITTED_RUSTC_DEPENDENCIES)), &[]),
```

Motivation is that I have no idea what the 5 args to the `new` fn is and would have to look at the `new` definition every single time 😆